botocore/regions.py

```diff
@@ -401,3 +402,34 @@ def _expand_template(
         return template.format(
             service=service_name, region=endpoint_name, dnsSuffix=dnsSuffix
         )
+
+
+class EndpointResolverBuiltins(str, Enum):
+    # The AWS Region configured for the SDK client (str)
+    AWS_REGION = "AWS::Region"
+    # Whether the UseFIPSEndpoint configuration option has been enabled for the
+    # SDK client (bool)
+    AWS_USE_FIPS = "AWS::UseFIPS"
+    # Whether the UseDualStackEndpoint configuration option has been enabled for
+    # the SDK client (bool)
+    AWS_USE_DUALSTACK = "AWS::UseDualStack"
+    # Whether the global endpoint should be used, rather the the regional
+    # endpoint for us-east-1 (bool)
+    AWS_STS_USE_GLOBAL_ENDPOINT = "AWS::STS::UseGlobalEndpoint"
+    # Whether the global endpoint should be used, rather then the regional
+    # endpoint for us-east-1 (bool)
+    AWS_S3_USE_GLOBAL_ENDPOINT = "AWS::S3::UseGlobalEndpoint"
+    # Whether S3 Transfer Acceleration has been requested (bool)
+    AWS_S3_ACCELERATE = "AWS::S3::Accelerate"
+    # Whether S3 Force Path Style has been enabled  (bool)
+    AWS_S3_FORCE_PATH_STYLE = "AWS::S3::ForcePathStyle"
+    # Whether to use the ARN region or raise an error when ARN and client
+    # region differ (for s3 service only)
+    AWS_S3_USE_ARN_REGION = "AWS::S3::UseArnRegion"
+    # Whether to use the ARN region or raise an error when ARN and client
+    # region differ (for s3-control service only).
+    AWS_S3CONTROL_USE_ARN_REGION = 'AWS::S3Control::UseArnRegion'
+    # Whether multi-region access points (MRAP) should be disabled.
```

Comment:
These are missing typing info in the comment. Given everything is either a string or bool, we could probably use inline typing for this class. We have precedent for this in the NamedTuples now.